ethcore/src/client/client.rs

```diff
@@ -1570,22 +1570,28 @@ impl Call for Client {
 			let schedule = machine.schedule(env_info.number);
 			Executive::new(&mut clone, &env_info, &machine, &schedule)
 				.transact_virtual(&tx, options())
+		};
+
+		let cond = |gas| {
+			exec(gas)
 				.ok()
 				.map(|r| r.exception.is_none())
+				.unwrap_or(false)
 		};
 
-		let cond = |gas| exec(gas).unwrap_or(false);
-
 		if !cond(upper) {
 			upper = max_upper;
 			match exec(upper) {
-				Some(false) => return Err(CallError::Exceptional),
-				None => {
+				Ok(v) => {
+					if v.exception.is_some() {
+						return Err(CallError::Exceptional(v.exception.unwrap()))
```

Comment:
Could you use an `if let Some(exception) = v.exception` statement to avoid `.unwrap()`?

Reply:
I had read in docs that `.unwrap()` can cause panic, but before I check with `.is_some()`, so thought that this is fine.
Sorry, it's not clear for me, how `.unwrap()` can be avoided with `if let Some(exception) = v.exception`... can you explain?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=132afa5ce895d6d8d46882ee4df7d904

Reply:
Thank you for example.